### DISCUSSION TITLE
How do I test in production without having to recommit?

### DISCUSSION THREAD

**Discussion by adamjabarry** [vercel/community#7418] [Help] [answered]
### Summary

Hi,
I am new(ish) to Vercel and I have just deployed my application to GitHub and then to Vercel. While it is working locally, it is unfortunately not working when I deploy it (I think there is an issue with a call to an API or something). How do I go about running tests (console logs in my javascript, etc) to figure out what the issue is? I don't want to have to contiuously make amendments to my local code, push it to GitHub, and then wait for it to deploy. I feel like that would be super time consuming and would also clog up my GitHub commits. How do people usually go about testing in this environment. Any help would be greatly appreciated. Thank you!

### Steps to Reproduce

This is more of a conceptual problem rather than an individual code problem. I am sure I will have another issue in production but not in development and I would like to figure out how to take the steps to tackled it. Thanks!

**Comment by amyegan** [comment, marked as answer]
Hey @adamjabarry! I usually use [Vercel CLI](https://vercel.com/docs/cli) to check builds on my local computer. 

The [`vercel dev`](https://vercel.com/docs/cli/dev#vercel-dev) command is useful for running a development build, and [`vercel build`](https://vercel.com/docs/cli/build#vercel-build) can be used to make sure a build will succeed. 

If the problem is caused by a missing environment variable, you can review them in project settings or [use the CLI to pull variables into a local file](https://vercel.com/docs/cli/env#exporting-development-environment-variables) and check that they match what you expect.

You can also [deploying projects from Vercel CLI](https://vercel.com/docs/cli/deploying-from-cli#deploying-projects-from-vercel-cli) if you want to deploy without pushing test code to GitHub.

**Reply by adamjabarry** [reply]
Hi Amy,

Thank you so much for your response. It has been super helpful!

Best,
Adam

[a comment by an automated account is not shown]
